qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -193,13 +269,12 @@ def get_qubit_operators(
         num_particles = getattr(problem, "num_particles", None)
 
         # 1. Convert the main operator (hamiltonian) to a Qubit Operator and apply two qubit reduction
-
         if isinstance(self.qubit_converter, QubitConverter):
             self.qubit_converter.force_match(num_particles=num_particles)
             main_op = self.qubit_converter.convert_only(main_operator, num_particles=num_particles)
+        elif isinstance(self.qubit_converter, TaperedQubitMapper):
+            main_op = self.qubit_converter.map_clifford(main_operator)
         else:
-            if isinstance(self.qubit_converter, ParityMapper):
-                self.qubit_converter.num_particles = num_particles
```

Comment:
After some thinking, the only use case is: a poorly defined ParityMapper with the wrong particle number used on a QEOM problem providing the "true" number of particles. We can keep it, I went too quickly on that maybe

Reply:
I think we need to decide on what our expectation for the mappers are (i.e. whether or not we expect our algorithms to potentially overwrite the information in them).
I would like to avoid hard-coding cases like this as much as possible. Let us remove it and if it becomes an issue down the road, we can either re-add it or improve the documentation on this aspect.